substrate/executor/src/native_executor.rs

```diff
@@ -28,12 +28,15 @@ use RuntimeInfo;
 
 // For the internal Runtime Cache:
 // Is it compatible enough to run this natively or do we need to fall back on the WasmModule
+
 enum Compatibility {
 	InvalidVersion(WasmModule),
-	IsCompatible(RuntimeVersion),
+	IsCompatible(RuntimeVersion, WasmModule),
```

Comment:
If natve+wasm execution is requested this should still cache the module. Wasm execution will be performed even if the version is compatible.